CONSOLE-3695: Add details item extension to console dynamic plugin SDK (#13240)
openshift/console: frontend/packages/console-dynamic-plugin-sdk/docs/console-extensions.md

---

## `console.resource/details-item`

### Summary 

Adds a new details item to the details page resource summary.

### Properties

| Name | Value Type | Optional | Description |
| ---- | ---------- | -------- | ----------- |
| `model` | `ExtensionK8sModel` | no | The resource kind to which the details item applies. |
| `id` | `string` | no | A unique details item identifier. |
| `column` | `DetailsItemColumn` | no | @defaultValue 'right' |
| `title` | `string` | no | The title of the details item. |
| `path` | `string` | yes | @example 'spec.template.spec.containers[0].image' |

> | `component` | `CodeRef<React.ComponentType<DetailsItemComponentProps<K8sResourceCommon>>>` | yes | The component to be rendered as the details item value. This takes precedence over the<br/>`path` property. |

Reviewer: Technically this is passive voice. You could do something like:
```suggestion
| `component` | `CodeRef<React.ComponentType<DetailsItemComponentProps<K8sResourceCommon>>>` | yes | The component will render as the details item value. This takes precedence over the<br/>`path` property. |
```
